common/src/main/java/com/microsoft/identity/common/nativeauth/internal/controllers/NativeAuthMsalController.kt

```diff
@@ -704,8 +709,17 @@ class NativeAuthMsalController : BaseNativeAuthController() {
         // other tenants. Those tokens will be 'sparse', meaning that their AT/RT will not be loaded
         val fullCacheRecord = cacheRecords[0]
 
-        if (accessTokenIsNull(fullCacheRecord)) {
-            throw ServiceException(ErrorStrings.NATIVE_AUTH_NO_ACCESS_TOKEN_FOUND, "No access token found during refresh - user must be signed out.", null)
+        if (!accessTokenIsNull(fullCacheRecord) && silentTokenCommandParameters.scopes == null) {
```

Comment:
Same as other comment, my view on how we can achieve re-usability is to use the existing `acquireTokenSilent()` and can remove this extra added logic.